src/main/java/com/iexec/core/chain/BlockchainConnectionHealthIndicator.java

```diff
@@ -0,0 +1,165 @@
+    }
+
+    /**
+     * Check blockchain is reachable by retrieving the latest block number.
+     * <p>
+     * If it isn't, then increment {@link BlockchainConnectionHealthIndicator#consecutiveFailures} counter.
+     * If counter reaches {@link BlockchainConnectionHealthIndicator#outOfServiceThreshold},
+     * then this Health Indicator becomes {@link Status#OUT_OF_SERVICE}.
+     * <p>
+     * If blockchain is reachable, then reset the counter.
+     * <p>
+     * /!\ If the indicator becomes {@code OUT_OF_SERVICE}, then it stays as is until the Scheduler is restarted
+     * even if the blockchain is later available again.
+     */
+    void checkConnection() {
+        final long latestBlockNumber = web3jService.getLatestBlockNumber();
+        if (latestBlockNumber == 0) {
+            connectionFailed();
+        } else {
+            connectionSucceeded();
+        }
+    }
+
+    /**
+     * Increment the {@link BlockchainConnectionHealthIndicator#consecutiveFailures} counter.
+     * <p>
+     * If {@link BlockchainConnectionHealthIndicator#outOfServiceThreshold} has been reached,
+     * then set {@link BlockchainConnectionHealthIndicator#outOfService} to {@code true}.
+     * <p>
+     * If first failure, set the {@link BlockchainConnectionHealthIndicator#firstFailure} to current time.
+     */
+    private void connectionFailed() {
+        ++consecutiveFailures;
+        if (consecutiveFailures >= outOfServiceThreshold) {
+            outOfService = true;
+            log.error("Blockchain hasn't been accessed for a long period. " +
+                    "This Scheduler is now OUT-OF-SERVICE until it is restarted." +
+                    " [unavailabilityPeriod:{}]", pollingInterval.multipliedBy(outOfServiceThreshold));
+        } else {
+            if (consecutiveFailures == 1) {
+                firstFailure = LocalDateTime.now(clock);
+            }
+            log.warn("Blockchain is unavailable. Will retry connection." +
+                            " [unavailabilityPeriod:{}, nextRetry:{}]",
+                    pollingInterval.multipliedBy(consecutiveFailures), pollingInterval);
+        }
+    }
+
+    /**
+     * Reset {@link BlockchainConnectionHealthIndicator#consecutiveFailures} to {@code 0}.
+     * <p>
+     * If never been OUT-OF-SERVICE, then:
+     * <ul>
+     *     <li>Reset the {@link BlockchainConnectionHealthIndicator#firstFailure} var to {@code null};</li>
+     *     <li>Log a "connection restored" message.</li>
+     * </ul>
+     */
+    private void connectionSucceeded() {
+        if (!outOfService) {
+            if (consecutiveFailures > 0) {
+                log.info("Blockchain connection is now restored after a period of unavailability." +
+                        " [unavailabilityPeriod:{}]", pollingInterval.multipliedBy(consecutiveFailures));
+            }
+            firstFailure = null;
+        }
+        consecutiveFailures = 0;
```

Comment:
Could it be simplified as follow ?
```suggestion
        if (!outOfService && consecutiveFailures > 0) {
            log.info("Blockchain connection is now restored after a period of unavailability." +
                    " [unavailabilityPeriod:{}]", pollingInterval.multipliedBy(consecutiveFailures));
            firstFailure = null;
            consecutiveFailures = 0;
        }
```

Reply:
This version wouldn't work. `consecutiveFailures` has to be reset every time the connection succeeds. However, the following would work:
```suggestion
        if (!outOfService && consecutiveFailures > 0) {
            log.info("Blockchain connection is now restored after a period of unavailability." +
                    " [unavailabilityPeriod:{}]", pollingInterval.multipliedBy(consecutiveFailures));
            firstFailure = null;
        }
        consecutiveFailures = 0;
```
I'll apply it as it simplifies our code. Thanks for the suggestion!